widget/form.go

```diff
@@ -47,13 +50,17 @@ type Form struct {
 	OnCancel   func()
 	SubmitText string
 	CancelText string
+	Validator  func(*Form) error
```

Comment:
Is there a way we can avoid this being form specific?
Something like the following would be really interesting:

`Form.Validator = validation.NewEqualText(email1, email2)`

A crass example but as you can see the `Form` is not passed in because it isn't really what is being validated...